### DISCUSSION TITLE
Can Ocelot perform runtime routing?

### DISCUSSION THREAD

**Discussion by lobbo232** [ThreeMammals/Ocelot#1882] [Q&A] [answered]
I am looking to see if Ocelot will fill a criteria for me.

I have an infrastructure with a Gateway API that talks to several other identical APIs that are located in different sites. I need to route to the correct API based on the current users session.

i.e., a request from User A needs to be routed to the API of Company A, and a request from User B needs to be routed to the API of Company B.

The API end points that the Gateway needs to route to would be stored in a connected database.

My question is; Am I able to use Ocelot to catch an incoming request, look up the API endpoint from my database in C#, and route that request to that end point?

I can't store all my endpoints in the json file as they'd need to be changeable without manually editing files.

**Comment by ggnaegi** [comment, marked as answer]
Hello, maybe with [delegating handlers](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html)?

I'm using an handler to modify some parameters in the downstream route, according to user's reference.

Yes, you can modify the request URI in the SendAsync method in DelegatingHandler

**Reply by raman-m** [reply]
Thanks @ggnaegi for answering the question!
In addition, the author can develop custom Load Balancer, right?
But reading data from/to DB is required, therefore a custom delegating handler is better approach, for sure.
